src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace.E2ETest/HyperspaceTests.cs

```diff
@@ -0,0 +1,133 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Microsoft.Spark.E2ETest.Utils;
+using Microsoft.Spark.Extensions.Hyperspace.Index;
+using Microsoft.Spark.Sql;
+using Xunit;
+
+namespace Microsoft.Spark.Extensions.Hyperspace.E2ETest
+{
+    /// <summary>
+    /// Test suite for Hyperspace index management APIs.
+    /// </summary>
+    [Collection(Constants.HyperspaceTestContainerName)]
+    public class HyperspaceTests
+    {
+        private readonly SparkSession _spark;
+        private readonly Hyperspace _hyperspace;
+
+        // Fields needed for sample DataFrame.
+        private readonly DataFrame _sampleDataFrame;
+        private readonly string _sampleIndexName;
+        private readonly IndexConfig _sampleIndexConfig;
+
+        public HyperspaceTests(HyperspaceFixture fixture)
+        {
+            _spark = fixture.SparkFixture.Spark;
+            _hyperspace = fixture.Hyperspace;
+
+            // Make sure that any leftover indexes are removed before starting a test.
+            Directory.Delete(fixture.HyperspaceSystemPath, true);
```

Comment:
Maybe creating a temporary directory per suite is cleaner instead of doing this at fixture? I think you can do `_spark.Conf.Set("spark.hyperspace.system.path", tmp_dir_loc)` to update.

Reply:
Changed.